docs/pipelineruns.md

```diff
@@ -362,6 +363,70 @@ The `timeout` value is a `duration` conforming to Go's
 
+## Monitoring execution status
+
+As your `PipelineRun` executes, its `status` field accumulates information on the execution of each `TaskRun`
+as well as the `PipelineRun` as a whole. This information includes the name of the pipeline `Task` associated
+to a `TaskRun`, the complete [status of the `TaskrRun`](taskruns.md#monitoring-execution-status) and details
+about `Conditions` that may be associated to a `TaskRun`.
+
+The following example shows an extract from tje `status` field of a `PipelineRun` that has executed successfully:
+
+```yaml
+completionTime: "2020-05-04T02:19:14Z"
+conditions:
+- lastTransitionTime: "2020-05-04T02:19:14Z"
+  message: 'Tasks Completed: 4, Skipped: 0'
+  reason: Succeeded
+  status: "True"
+  type: Succeeded
+startTime: "2020-05-04T02:00:11Z"
+taskRuns:
+  triggers-release-nightly-frwmw-build-ng2qk:
+    pipelineTaskName: build
+    status:
+      completionTime: "2020-05-04T02:10:49Z"
+      conditions:
+      - lastTransitionTime: "2020-05-04T02:10:49Z"
+        message: All Steps have completed executing
+        reason: Succeeded
+        status: "True"
+        type: Succeeded
+      podName: triggers-release-nightly-frwmw-build-ng2qk-pod-8vj99
+      resourcesResult:
+      - key: commit
+        resourceRef:
+          name: git-source-triggers-frwmw
+        value: 9ab5a1234166a89db352afa28f499d596ebb48db
+      startTime: "2020-05-04T02:05:07Z"
+      steps:
+      - container: step-build
+        imageID: docker-pullable://golang@sha256:a90f2671330831830e229c3554ce118009681ef88af659cd98bfafd13d5594f9
+        name: build
+        terminated:
+          containerID: docker://6b6471f501f59dbb7849f5cdde200f4eeb64302b862a27af68821a7fb2c25860
+          exitCode: 0
+          finishedAt: "2020-05-04T02:10:45Z"
+          reason: Completed
+          startedAt: "2020-05-04T02:06:24Z"
+  ```
+
+The following tables shows how to read the overall status of a `PipelineRun`:
+
+`status`|`reason`|`completionTime` is set|PipelineRun status
+:-------|:-------|:---------------------:|--------------:
+Unknown|Started|No|The `PipelineRun` has just been picked up by the controller.
+Unknown|Running|No|The `PipelineRun` has been validate and started to perform its work.
+Unknown|PipelineRunCancelled|No|The user requested the PipelineRun to be cancelled. Cancellation has not be done yet.
+True|Succeeded|Yes|The `PipelineRun` completed successfully.
+False|Failed|Yes|The `PipelineRun` failed because one of the `TaskRuns` failed.
+False|\[Error message\]|No|The `PipelineRun` encountered an error, but it's still running.
+False|\[Error message\]|Yes|The `PipelineRun` failed with a permanent error (usually validation).
+False|PipelineRunCancelled|Yes|The `PipelineRun` was cancelled successfully.
+False|PipelineRunTimeout|Yes|The `PipelineRun` timed out.
+
+When a `PipelineRun` changes status, [events](events.md#pipelineruns) are triggered accordingly.
+
```

Comment:
thanks @afrittoli for the table, its very helpful and easier to understand, few comments here:

1) it will be easier to understand `PipelineRunCancelled` if both the rows with that reason are in sequence.
2) We are missing `Completed` with status `True` in this table. I think we should include it since this table represents the overall status of `PipelineRun`.
3) May be rename column heading `PipelineRun Status` to description or something similar.
4) This is misleading "The PipelineRun encountered an error, but it's still running." This is happening because [PipelineRun Status](https://github.com/tektoncd/pipeline/blob/be9ac3bf6b27f6beb3f4b7c828ef8388f9b323c3/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L437) is updated with one failure without waiting for all the tasks in `nextRprts` to finish. We can add little more to this description until we fix issue #1680, "The PipelineRun encountered an error, but waiting for already running tasks to finish."

Reply:
> thanks @afrittoli for the table, its very helpful and easier to understand, few comments here:
> 
>     1. it will be easier to understand `PipelineRunCancelled` if both the rows with that reason are in a sequence.

They are sorted by status. I could sort them by reason, that would make it more readable from a Reason pov, less readable from a status pov.... not sure which is best.
I thought status should go first as it's the primary indicator of the condition.

> 
>     2. We are missing `Completed` with status `True` in this table. I think we should include it since this table represents the overall status of `PipelineRun`.
> 
Nice catch, thank you

>     3. May be rename column heading `PipelineRun Status` to description or something similar.

Fixed
> 
>     4. This is misleading "The PipelineRun encountered an error, but it's still running." This is happening because [PipelineRun Status](https://github.com/tektoncd/pipeline/blob/be9ac3bf6b27f6beb3f4b7c828ef8388f9b323c3/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L437) is updated with one failure without waiting for all the tasks in `nextRprts` to finish. We can add little more to this description until we fix issue #1680, "The PipelineRun encountered an error, but waiting for already running tasks to finish."

That's not the case I was referring to here. 
If any non-permanent error happens during reconcile, the reason is set to the error description, the status is set to failed, but it may be switch back to succeeded if the next reconcile cycle clears the error.

Reply:
Uhm, I checked this again, and there does not seem to be any transient error reported by the pipelinerun controller, so I will remove this one.
I think I will add the point you made as a general notice to the pipeline run docs.